rasa/nlu/regex_message_handler.py

```diff
@@ -0,0 +1,292 @@
+        otherwise). The new message is created on purpose to get rid of all attributes
+        that NLU components might have added based on the `TEXT` attribute which
+        does not contain real text but the regex we expect here.
+
+        Args:
+          messages: list of messages
+          domain: the domain
+        Returns:
+           list of messages where the i-th message is equal to the i-th input message
+           if that message does not need to be unpacked, and a new message with the
+           extracted attributes otherwise
+        """
+        return [self._unpack(message, domain) for message in messages]
+
+    def _unpack(self, message: Message, domain: Domain) -> Message:
+        """Unpacks the messsage if `TEXT` contains an encoding of attributes.
+
+        Args:
+          message: some message
+          domain: the domain
+        Returns:
+           the given message if that message does not need to be unpacked, and a new
+           message with the extracted attributes otherwise
+        """
+        user_text = message.get(TEXT).strip()
+
+        # If the prefix doesn't match, we don't even need to try to match the pattern.
+        if not user_text.startswith(self._prefix):
+            return message
+
+        # Try to match the pattern.
+        match = self._pattern.match(user_text)
+
+        # If it doesn't match, then (potentially) something went wrong, because the
+        # message text did start with the special prefix -- however, a user might
+        # just have decided to start their text this way.
+        if not match:
+            logger.warning(f"Failed to parse intent end entities from '{user_text}'.")
+            return message
+
+        # Extract attributes from the match - and validate it via the domain.
+        intent_name = RegexMessageHandler._parse_intent_name(match, domain)
+        confidence = RegexMessageHandler._parse_optional_confidences(match)
+        entities = RegexMessageHandler._parse_optional_entities(match, domain)
+
+        # The intent name is *not* optional, but during parsing we might find out
+        # that the given intent is unknown (and warn). In this case, stop here.
+        if intent_name is None:
+            return message
+
+        if match.group("rest"):
+            rasa.shared.utils.io.raise_warning(
+                f"Failed to parse arguments in line '{match.string}'. "
+                f"Failed to interpret some parts. "
+                f"Continuing without {match.group('rest')}. ",
+                docs=DOCS_URL_STORIES,
+            )
+
+        # Add the results to the message.
+        # NOTE: In the original implementation the message TEXT was set to the
+        # text from which the regex was extracted. However, if the regex matches
+        # then we know there is *no* text included in that "TEXT".
+        # TODO: .. is `output_properties` (cf. Message) even used anywhere?
```

Comment:
I don't think it's used now but again, `Message` is a part of `shared` page so it might be in use by third-party apps.

Reply:
I can add the text back in - but then all models working "end to end" will see that weird expression? 🤔  Whom could we ask?

Reply:
@JEM-Mosig this might have to do something with #9587 - in addition to TED just being TED and being "only_e2e" in that case ... what do you think?

Reply:
> I can add the text back in - but then all models working "end to end" will see that weird expression? 🤔 

Came to the conclusion [here](https://github.com/RasaHQ/rasa/issues/9587#issuecomment-916061233) that for various reasons (e.g. how the (only) ensemble works) models shouldn't see those messages -- if and only if a `RulePolicy` is contained in the ensemble. So looks to me like it would be better to remove this here.